src/sentry/static/sentry/app/bootstrap.jsx

```diff
@@ -35,6 +35,9 @@ Sentry.init({
     new Tracing({
       tracingOrigins: ['localhost', 'sentry.io', /^\//],
     }),
+    new TransactionActivity({
+      idleTimeout: 500,
```

Comment:
How does this timeout work?

Reply:
Exactly, and after 500ms of no activity, the transaction will be finished and therefore flushed.
Also the we take the end timestamp of the last completed span in order to have the correct timing.
All of this happens in the SDK now.